pkg/operator/operator.go

```diff
@@ -441,26 +439,31 @@ func (o *Operator) sync(key string) error {
 	}
 	factory := manifests.NewFactory(o.namespace, o.namespaceUserWorkload, config, o.loadInfrastructureConfig(), proxyConfig, o.assets)
 
+	// update prometheus-operator before anything else
```

Comment:
Maybe move this right before the first `tasks.NewTaskGroup`?

```suggestion
	// Update prometheus-operator before anything else because it is responsible for managing many other resources (e.g. Prometheus, Alertmanager, Thanos Ruler, ...).
```